File: backend/app/schemas/statistics.py
from pydantic import BaseModel


class OverviewOut(BaseModel):
    total_cases: int
    total_violations: int
    total_reports: int = 0
    approve_rate: float
    reject_rate: float
    pending_count: int
    today_new: int = 0
    notify_success_rate: float = 0.0


class ByLocationItem(BaseModel):
    name: str | None
    value: int


class ByLocationOut(BaseModel):
    items: list[ByLocationItem]


class ByTypeItem(BaseModel):
    name: str
    value: int
    percentage: float = 0.0


class ByTypeOut(BaseModel):
    items: list[ByTypeItem]


class ByTimeItem(BaseModel):
    date: str
    count: int


class ByTimeOut(BaseModel):
    items: list[ByTimeItem]

File: smart-traffic-frontend/src/types/statistics.ts
/** 概览统计 */
export interface StatisticsOverview {
  total_cases: number
  total_violations: number
  approve_rate: number
  reject_rate: number
  pending_count: number
  today_new: number
}

/** 按类型统计项 */
export interface ViolationTypeItem {
  name: string
  value: number
}

/** 按区域统计项 */
export interface LocationItem {
  name: string
  value: number
}

/** 趋势数据点 */
export interface TrendPoint {
  date: string
  count: number
}

/** 路段时段热点 */
export interface RoadTimeHotspot {
  road: string
  time_slot: string
  count: number
}

/** 统计大屏完整响应 */
export interface StatisticsPayload {
  overview: StatisticsOverview
  violation_types: ViolationTypeItem[]
  locations: LocationItem[]
  trend?: TrendPoint[]
  road_time_hotspots?: RoadTimeHotspot[]
}
